awscrt/http.py

```diff
@@ -219,6 +249,33 @@ def request(self, request, on_response=None, on_body=None):
         return HttpClientStream(self, request, on_response, on_body)
 
 
+class Http2ClientConnection(HttpClientConnection):
+    """
+    HTTP/2 client connection.
+
+    This class extends HttpClientConnection with HTTP/2 specific functionality.
+    """
+    @classmethod
+    def new(cls,
```

Comment:
debatable: not sure it's worth giving this an explicit constructor? It's not doing anything to actually *cause* the user to get an HTTP/2 connection, like setting alpn=h2 or prior_knowledge_http2. It's just forcing an error if the user didn't end up getting an HTTP/2 connection.

For better or worse, our API in C takes all possible options, because it's possible to establish a connection and not know what you're going to get.